the_wizz/pair_maker.py

```diff
@@ -99,22 +103,23 @@ def run(self, reference_catalog, unknown_catalog, use_unkn_weights=False):
         Parameters
         ----------
         reference_catalog : 'dict' of `numpy.ndarray`
-            Dictionary containing arrays 
+            Dictionary containing arrays
         """
-        unkn_vects = self._convert_radec_to_xyz(
-            np.radians(unknown_catalog["ra"]),
-            np.radians(unknown_catalog["dec"]))
-        unkn_tree = cKDTree(unkn_vects)
+        unkn_tree = SphericalKDTree(
+            unknown_catalog["ra"], unknown_catalog["dec"])
+        unkn_vects = unkn_tree.tree.data
         unkn_ids = unknown_catalog["id"]
 
         redshifts = reference_catalog["redshift"]
         z_mask = np.logical_and(redshifts > self.z_min, redshifts < self.z_max)
         ref_ids = reference_catalog["id"][z_mask]
-        ref_vects = self._convert_radec_to_xyz(
-            np.radians(reference_catalog["ra"][z_mask]),
-            np.radians(reference_catalog["dec"][z_mask]))
         redshifts = reference_catalog["redshift"][z_mask]
-        dists = self.distance_metric(redshifts).value
+        try:
+            dists = self.distance_metric(redshifts).value
+        except AttributeError:
+            dists = self.distance_metric(redshifts)
```

Comment:
Not sure why we would need this. I specify `distance_metric` as a cosmology object from astropy and that is what it is assumed to be later in the code.